### PR TITLE
[No Ticket] Add settings, flake and pytest to osf-pigeon

### PR DESCRIPTION
**Pull request by Johnetordoff**
## Purpose

Upload a series of nested directories to to Internet Archive

## Changes

- add script with tests for IA upload

## QA Notes

- not user facing, no migrations 

## Documentation

Not user facing.

## Side Effects

None that I know of.

## Ticket

https://openscience.atlassian.net/browse/ENG-1159

### CODE REVIEW

File: IA/IA_upload.py [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+                size = len(data)
+                if size > CHUNK_SIZE:
+                    tasks.append(chunked_upload(bucket_name, path, data))
+                else:
+                    tasks.append(upload(bucket_name, path, data))
+
+    await asyncio.gather(*tasks)
+
+
+async def upload(bucket_name: str, filename: str, file_content: bytes):
```
Comment (corbinSanders) [inline IA/IA_upload.py]: As we discussed in flowdock, not sure if a regular upload will be necessary. It depends on the max size of a non-chunk upload.

File: IA/tests/test_upload.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+    @responses.activate
+    def test_IA_upload(self):
+        responses.add(
+            responses.Response(
+                responses.PUT,
+                'http://s3.us.archive.org/bucketname/file_name',
+            )
+        )
+        asyncio.run(upload('bucketname', 'file_name', b'content'))
```
Comment (corbinSanders) [inline IA/tests/test_upload.py]: You probably have already looked into this, but would there be a better way to mock these? It looks like the only assertions these do is making sure it doesn't error out.

File: IA/IA_upload.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+
+def mp_from_ids(mp_id: str, mp_keyname: str, bucket: str) -> MultiPartUpload:
+    mp = MultiPartUpload(bucket)
+    mp.key_name = mp_keyname
+    mp.id = mp_id
+    return mp
+
+
+async def gather_and_upload(bucket_name: str, parent: str):
```
Comment (corbinSanders) [inline IA/IA_upload.py]: More Py3 💪

**Review by corbinSanders — CHANGES_REQUESTED** [review]

File: IA/tests/test_upload.py (hunk 2 above)
Comment (Johnetordoff) [reply IA/tests/test_upload.py]: Yeah it's pretty difficult to mock.

File: settings/locals.py [hunk 4]
```diff
@@ -0,0 +1 @@
+OSF_API_URL = 'https://localhost:8000/'
```
Comment (brianjgeiger) [inline settings/locals.py]: Should this file be ignored? It looks like it's meant to be.

**Review by brianjgeiger — CHANGES_REQUESTED** [review]
Only one small gitignore problem, otherwise looks good.

**Review by brianjgeiger — APPROVED** [review]

**Pull request merged**